### PR TITLE
feat(objects): support all issue statistics endpoints

### PR DESCRIPTION
**Pull request by nejch**
Closes https://github.com/python-gitlab/python-gitlab/issues/1261

### CODE REVIEW

File: docs/gl_objects/issues.rst [hunk 1]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+  + :class:`gitlab.v4.objects.ProjectIssuesStatisticsManager`
+  + :attr:`gitlab.v4.objects.Project.issuesstatistics`
+
+
+* GitLab API: https://docs.gitlab.com/ce/api/issues_statistics.htm
+
+Examples
+---------
+
+Get statistics of all issues the user has access to::
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:253]: The docs say that it returns by default the issues created by the current user.

https://docs.gitlab.com/ee/api/issues_statistics.html#get-issues-statistics

```
By default it returns only issues created by the current user.
```

File: docs/gl_objects/issues.rst [hunk 2]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+
+    statistics = gl.issuesstatistics.get()
+
+Get statistics of issues for the user with ``foobar`` in ``title`` and ``description``::
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:257]: s/and/or/

```
Get statistics of issues for the user with ``foobar`` in the ``title`` or the ``description``::
```

File: docs/gl_objects/issues.rst [hunk 3]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+
+    statistics = gl.issuesstatistics.get(search='foobar')
+
+Get statistics of all issues in a group::
+
+    statistics = group.issuesstatistics.get()
+
+Get statistics of issues in a group with ``foobar`` in ``title`` and ``description``::
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:265]: s/and/or/

```
Get statistics of issues in a group with ``foobar`` in the ``title`` or the ``description``::
```

File: docs/gl_objects/issues.rst [hunk 4]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+
+    statistics = group.issuesstatistics.get(search='foobar')
+
+Get statistics of all issues in a project::
+
+    statistics = project.issuesstatistics.get()
+
+Get statistics of issues in a project with ``foobar`` in ``title`` and ``description``::
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:273]: s/and/or/

```
Get statistics of issues in a project with ``foobar`` in the ``title`` or the ``description``::
```

File: docs/gl_objects/issues.rst [hunk 5]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+=========================
+
+Reference
+---------
+
+* v4 API:
+
+  + :class:`gitlab.v4.objects.IssuesStatistics`
+  + :class:`gitlab.v4.objects.IssuesStatisticsManager`
+  + :attr:`gitlab.issuesstatistics`
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:239]: Is it possible to do: `gitlab.issues_statistics` ?

Changing `issuesstatistics` to `issue_statistics`

File: docs/gl_objects/issues.rst [hunk 6]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+  + :class:`gitlab.v4.objects.GroupIssuesStatistics`
+  + :class:`gitlab.v4.objects.GroupIssuesStatisticsManager`
+  + :attr:`gitlab.v4.objects.Group.issuesstatistics`
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:242]: Is it possible to do: `gitlab.v4.objects.Group.issues_statistics` ?

Changing `issuesstatistics` to `issue_statistics`

File: docs/gl_objects/issues.rst [hunk 7]
```diff
@@ -225,3 +225,51 @@ Link issue ``i1`` to issue ``i2``::
...
+  + :class:`gitlab.v4.objects.ProjectIssuesStatistics`
+  + :class:`gitlab.v4.objects.ProjectIssuesStatisticsManager`
+  + :attr:`gitlab.v4.objects.Project.issuesstatistics`
```
Comment (JohnVillalovos) [inline docs/gl_objects/issues.rst:245]: Is it possible to do: `gitlab.v4.objects.Project.issues_statistics` ?

Changing `issuesstatistics` to `issue_statistics`

File: gitlab/v4/objects/statistics.py [hunk 8]
```diff
@@ -19,6 +23,25 @@ class ProjectAdditionalStatisticsManager(GetWithoutIdMixin, RESTManager):
     _from_parent_attrs = {"project_id": "id"}
 
 
+class IssuesStatistics(RefreshMixin, RESTObject):
+    _id_attr = None
+
+
+class IssuesStatisticsManager(GetWithoutIdMixin, RESTManager):
+    _path = "/issues_statistics"
+    _obj_cls = IssuesStatistics
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/statistics.py:32]: When are `_optional_get_attrs` used? I see we have code for it in the CLI but only one class seems to use it ( `LicenseManager` )

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

File: docs/gl_objects/issues.rst (hunk 1 above)
Comment (nejch) [reply docs/gl_objects/issues.rst:257]: Thanks, done!

File: docs/gl_objects/issues.rst (hunk 5 above)
Comment (nejch) [reply docs/gl_objects/issues.rst]: I've changed this and for now silently deprecated the existing `issuesstatistics` in Project. I guess we'd need a nice helper to make it easy to add deprecation warnings to any manager class or attribute.

**Review by JohnVillalovos — APPROVED** [review]
LGTM (Looks Good To Me).

Thanks!

File: gitlab/v4/objects/statistics.py (hunk 8 above)
Comment (nejch) [reply gitlab/v4/objects/statistics.py:32]: Realized I accidentally resolved this earlier. They are explicitly displayed with the CLI help if given, e.g.:

```shell
(.venv) nejc@nuc:~/repos/python-gitlab$ gitlab license get --help
usage: gitlab license get [-h] [--sudo SUDO] --key KEY [--project PROJECT] [--fullname FULLNAME]

optional arguments:
  -h, --help           show this help message and exit
  --sudo SUDO
  --key KEY
  --project PROJECT
  --fullname FULLNAME
```

Not sure how useful this is ATM, I'd wait for some CLI rework to consider this (it's another manual list to maintain).

Comment (JohnVillalovos) [reply gitlab/v4/objects/statistics.py:32]: Thanks for the info.

**Pull request merged**